### PR TITLE
add py3 compatibility to pyethapp

### PR DESCRIPTION
**Pull request by changwu-tw**
Make pyethapp work both on py2/py3.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -30,31 +30,19 @@ def run_tests(self):
...
-            requirement.strip(),
-            requirement.strip(),
-        )
-
-        INSTALL_REQUIRES.append(dependency)
-
-INSTALL_REQUIRES = list(set(INSTALL_REQUIRES))
-
-DEPENDENCY_LINKS = []
-if os.environ.get("USE_PYETHEREUM_DEVELOP"):
```
Comment (hwwhww) [inline setup.py:50]: If `USE_PYETHEREUM_DEVELOP` flag has been removed, we also need to remove `USE_PYETHEREUM_DEVELOP=1` in `.travis.yml`.

File: requirements.txt [hunk 2]
```diff
@@ -12,10 +12,11 @@ ipython>=5.4.0,<6.0.0
 statistics
 requests
 rlp>=0.5.1,<0.6.0
-devp2p>=0.8.0
```
Comment (hwwhww) [inline requirements.txt]: It seems that `devp2p` is still required if we use `python setup.py install` to install pyethapp?

**Comment by hwwhww** [comment]
In Python3, if we restart the daemon, the type of `network_id` in the DB is Python2 str so it raises an exception.
```
Traceback (most recent call last):
  File "/usr/local/bin/pyethapp", line 11, in <module>
    load_entry_point('pyethapp==1.5.0', 'console_scripts', 'pyethapp')()
  File "/usr/local/lib/python3.6/site-packages/click-6.7-py3.6.egg/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click-6.7-py3.6.egg/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.6/site-packages/click-6.7-py3.6.egg/click/core.py", line 1066, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.6/site-packages/click-6.7-py3.6.egg/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.6/site-packages/click-6.7-py3.6.egg/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/click-6.7-py3.6.egg/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pyethapp-1.5.0-py3.6.egg/pyethapp/app.py", line 232, in run
    service.register_with_app(app)
  File "/usr/local/lib/python3.6/site-packages/devp2p-0.9.2-py3.6.egg/devp2p/service.py", line 47, in register_with_app
    s = klass(app)
  File "/usr/local/lib/python3.6/site-packages/pyethapp-1.5.0-py3.6.egg/pyethapp/eth_service.py", line 150, in __init__
    self.config['data_dir'], db_network_id, sce['network_id']
RuntimeError: The database in '/root/.config/pyethapp' was initialized with network id bytearray(b'1') and can not be used when connecting to network id 1. Please choose a different data directory.
```


I think we can use `ethereum.utils.to_string` to fix it.

```
--- a/pyethapp/eth_service.py
+++ b/pyethapp/eth_service.py
@@ -34,7 +34,10 @@ from ethereum.slogging import get_logger
 from ethereum.exceptions import InvalidTransaction, InvalidNonce, \
     InsufficientBalance, InsufficientStartGas, VerificationFailed
 from ethereum.transactions import Transaction
-from ethereum.utils import encode_hex
+from ethereum.utils import (
+    encode_hex,
+    to_string,
+)

 from .synchronizer import Synchronizer
 from . import eth_protocol
@@ -142,8 +145,8 @@ class ChainService(WiredService):
             self.db.put("I am not pruning", "1")

         if 'network_id' in self.db:
-            db_network_id = self.db.get('network_id')
-            if db_network_id != str(sce['network_id']):
+            db_network_id = self.db.get(b'network_id')
+            if db_network_id != to_string(sce['network_id']):
                 raise RuntimeError(
                     "The database in '{}' was initialized with network id {} and can not be used "
                     "when connecting to network id {}. Please choose a different data directory.".format(
@@ -152,7 +155,7 @@ class ChainService(WiredService):
                 )

         else:
-            self.db.put('network_id', str(sce['network_id']))
+            self.db.put(b'network_id', to_string(sce['network_id']))
             self.db.commit()
```

File: requirements.txt (hunk 2 above)
Comment (changwu-tw) [reply requirements.txt]: Ahh, you're right!

File: setup.py (hunk 1 above)
Comment (changwu-tw) [reply setup.py:50]: Will remove the line in .travis.yml.

**Pull request closed without merging**